targettests/Remote-Sim/state_init_vector.cpp

```diff
@@ -0,0 +1,213 @@
+/*******************************************************************************
+ * Copyright (c) 2022 - 2024 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+
+// REQUIRES: remote-sim
+
+// clang-format off
+// RUN: nvq++ %cpp_std --enable-mlir --target remote-mqpu %s -o %t  && %t | FileCheck %s
+// RUN: nvq++ %cpp_std --enable-mlir --target remote-mqpu -fkernel-exec-kind=2 %s -o %t && %t | FileCheck %s
+// clang-format on
+
+#include <cudaq.h>
+#include <iostream>
+
+__qpu__ void test_complex_constant_array() {
+   cudaq::qvector v(std::vector<cudaq::complex>({ M_SQRT1_2, M_SQRT1_2, 0., 0.}));
+}
+
+__qpu__ void test_complex_constant_array2() {
+   cudaq::qvector v1(std::vector<cudaq::complex>({ M_SQRT1_2, M_SQRT1_2, 0., 0.}));
+   cudaq::qvector v2(std::vector<cudaq::complex>({ 0., 0., M_SQRT1_2, M_SQRT1_2}));
+}
+
+__qpu__ void test_complex_constant_array3() {
+   cudaq::qvector v({
+    cudaq::complex(M_SQRT1_2),
+    cudaq::complex(M_SQRT1_2),
+    cudaq::complex(0.0),
+    cudaq::complex(0.0)
+  });
+}
+
+__qpu__ void test_complex_array_param(std::vector<cudaq::complex> inState) {
+  cudaq::qvector q1 = inState;
+}
+
+__qpu__ void test_real_constant_array() {
```

Comment:
Would it be possible to make some bigger tests here, say, with 20-25 qubits?

Reply:
Hmm let me think how to best generate the constant that big:)

Reply:
Can it be generated in your test? I.e. (untested)
```cpp
std::vector<std::complex<double>> myVec(1<<20); 
for (int i = 0; i < myVec.size(); i++)
  myVec[i] = 1.0 / myVec.size();
```

Reply:
I gave it a try - please let me know if you have any suggestions. Max number of qubits I could use was 19 (20 caused a seg fault on the server)

Reply:
> I gave it a try - please let me know if you have any suggestions. Max number of qubits I could use was 19 (20 caused a seg fault on the server)

20 qubits sounds like the number of qubits where the state vector approaches ~8MB, which is near the stack size. I don't think we should be putting the state onto the stack. Should we malloc/free it?

Reply:
Added a test with 19 qubits.

Reply:
Removed for now, will send another PR adding the test and the error during synthesis.